packages/flutter_tools/lib/src/commands/build_setup_split_aot.dart

```diff
@@ -0,0 +1,136 @@
+import '../globals.dart' as globals;
+import '../project.dart';
+import '../reporting/reporting.dart';
+import '../runner/flutter_command.dart' show FlutterCommandResult;
+import 'assemble.dart';
+import 'build.dart';
+
+class BuildSetupSplitAotCommand extends BuildSubCommand {
+  BuildSetupSplitAotCommand({bool verboseHelp = false}) {
+    addTreeShakeIconsFlag();
+    usesTargetOption();
+    addBuildModeFlags(verboseHelp: verboseHelp);
+    usesFlavorOption();
+    usesPubOption();
+    usesBuildNumberOption();
+    usesBuildNameOption();
+    addSplitDebugInfoOption();
+    addDartObfuscationOption();
+    usesDartDefineOption();
+    usesExtraFrontendOptions();
+    addBundleSkSLPathOption(hide: !verboseHelp);
+    addEnableExperimentation(hide: !verboseHelp);
+    addBuildPerformanceFile(hide: !verboseHelp);
+    addNullSafetyModeOptions(hide: !verboseHelp);
+    argParser
+      ..addFlag('split-per-abi',
+        negatable: false,
+        help: 'Whether to split the APKs per ABIs. '
+              'To learn more, see: https://developer.android.com/studio/build/configure-apk-splits#configure-abi-split',
+      )
+      ..addMultiOption('target-platform',
+        splitCommas: true,
+        defaultsTo: <String>['android-arm', 'android-arm64', 'android-x64'],
+        allowed: <String>['android-arm', 'android-arm64', 'android-x86', 'android-x64'],
+        help: 'The target platform for which the app is compiled.',
+      );
+    usesTrackWidgetCreation(verboseHelp: verboseHelp);
+  }
+
+  @override
+  final String name = 'setupsplitaot';
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => <DevelopmentArtifact>{
+    DevelopmentArtifact.androidGenSnapshot,
+  };
+
+  @override
+  final String description = 'Build an Android split AAB file from your app.\n\n'
+    "This command can build debug and release versions of your application. 'debug' builds support "
+    "debugging and a quick development cycle. 'release' builds don't support debugging and are "
+    'suitable for deploying to app stores.';
+
+  @override
+  Future<Map<CustomDimensions, String>> get usageValues async {
+    final Map<CustomDimensions, String> usage = <CustomDimensions, String>{};
+
+    usage[CustomDimensions.commandBuildApkTargetPlatform] =
+        stringsArg('target-platform').join(',');
+    usage[CustomDimensions.commandBuildApkSplitPerAbi] =
+        boolArg('split-per-abi').toString();
+
+    if (boolArg('release')) {
+      usage[CustomDimensions.commandBuildApkBuildMode] = 'release';
+    } else if (boolArg('debug')) {
+      usage[CustomDimensions.commandBuildApkBuildMode] = 'debug';
+    } else if (boolArg('profile')) {
+      usage[CustomDimensions.commandBuildApkBuildMode] = 'profile';
+    } else {
+      // The build defaults to release.
+      usage[CustomDimensions.commandBuildApkBuildMode] = 'release';
+    }
+    return usage;
+  }
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
```

Comment:
we could probably add a helper to directly invoke the assemble code without an extra re-entrant call. Not 100% necessary but would be a good cleanup